### ISSUE TITLE
Move to alpine linux where possible for docker images

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#1430]
We should probably do this slowly to see if we like it, but I think it's worth at least considering. The main advantage of Alpine Linux for us is that it's super, ridiculously small, so uploading and downloading images becomes faster and easier. It should be fairly easy to test and if we like it, it should be easy to swap in. I had been putting off looking at it for awhile, but I finally did and I was heartened by the fact that the documentation is *tiny*. For an entire operating system, it's barely got anything, and not because it needs more. Because there's just not much to actually write about! 

I did a small test of alpine linux for our python3 image. It didn't work because there's no package for tesseract-ocr-eng, but that's getting moved into another package soon anyway, so once that happens, we can use alpine for our python image, and use Ubuntu or manual compilation for our BTE image. 

Anyway, more investigation is needed, and this one is a bit aspirational, but I'm putting it down in case somebody wants to start looking into it and playing with it. Like I say, we don't get a ton out of it, but it could still be a nice upgrade for devs.

**Comment by drewsilcock** [comment]
I'd be wary of using Alpine for anything Python related, this article is really at explaining why: https://pythonspeed.com/articles/alpine-docker-python/

tl;dr - Alpine uses musl instead of glibc

(I've encountered the stack size issue before and it took me *ages* to figure out that the problem was Alpine.)

**Comment by mlissner** [comment]
OK, that's terrible. To hell with that.
